library/core/src/time.rs

```diff
@@ -31,15 +32,13 @@ const MICROS_PER_SEC: u64 = 1_000_000;
 
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[repr(transparent)]
-#[rustc_layout_scalar_valid_range_start(0)]
-#[rustc_layout_scalar_valid_range_end(999_999_999)]
-struct Nanoseconds(u32);
+struct Nanoseconds(Ranged<u32, { 0..=(NANOS_PER_SEC as u128 - 1) }>);
```

Comment:
was there a reason for keeping the struct here and not in `unix/time.rs` ?

Reply:
Heh good point. I did this for another struct, not sure why I forgot for this one

Reply:
Better to remove the extraneous struct imo.